src/aws_lambda_utils/tracing_request_id_layer.rs

```diff
@@ -44,19 +48,95 @@ impl<S> Filter<S> for RequestIdLayer {
     }
 }
 
+#[derive(Default)]
 struct RequestIdVisitor {
     request_id: Option<String>,
+    aws_trace_id: Option<AwsTraceId>,
 }
 
 impl Visit for RequestIdVisitor {
     fn record_str(&mut self, field: &Field, value: &str) {
-        if field.name() == "request_id" && self.request_id.is_none() {
+        if field.name() == "request_id" && self.request_id.is_none() && !value.is_empty() {
             self.request_id = Some(value.to_string());
         }
+        if field.name() == "aws_trace_id" && self.aws_trace_id.is_none() && !value.is_empty() {
+            if let Ok(aws_trace_id) = AwsTraceId::from_str(value) {
+                self.aws_trace_id = Some(aws_trace_id);
+            }
+        }
     }
 
     fn record_debug(&mut self, _field: &tracing::field::Field, _value: &dyn std::fmt::Debug) {}
 }
 
 #[derive(Clone)]
-pub struct RequestId(pub String);
+pub(crate) struct RequestIds {
+    pub request_id: String,
+    pub aws_trace_id: Option<AwsTraceId>,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct AwsTraceId {
+    pub trace_id: String,
+    pub parent_id: Option<XRayId>,
+}
+
+impl FromStr for AwsTraceId {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut id = None;
+        let mut parent = None;
+        for x in s.split(';') {
+            if let Some((before, after)) = x.split_once('=') {
+                if before == "Root" {
+                    id = Some(after.to_string());
+                }
+                if before == "Parent" {
+                    parent = Some(after.to_string());
+                }
+            } else {
+                return Err("invalid".to_string());
```

Comment:
Does "invalid" have some special meaning here?

Reply:
I guess I was just bad at giving an error message when an invalid AWS-trace-id is seen. I'll update the error message to be nicer